scripts/build-augmentations.mjs

```diff
@@ -0,0 +1,47 @@
+import { parse } from '@babel/parser';
+import { traverse } from '@babel/core';
+import { readFileSync, writeFileSync } from 'fs';
+import { resolve } from 'path';
+import { createRequire } from 'module';
+import { fileURLToPath } from 'url';
+
+const require = createRequire(import.meta.url);
+const ts = require.resolve('typescript');
+const libDomFile = resolve(ts, '..', 'lib.dom.d.ts');
+const source = readFileSync(libDomFile).toString();
+const ast = parse(source, {
+  plugins: [['typescript', { dts: true }]],
+});
+
+const elements = new Set();
+traverse(ast, {
+  TSInterfaceDeclaration: function (path) {
+    if (path.node.id.name === 'HTMLElementTagNameMap') {
+      const items = path.node.body.body;
+      for (const item of items) {
+        elements.add(item.typeAnnotation.typeAnnotation.typeName.name);
+      }
+    }
+    if (path.node.id.name === 'SVGElementTagNameMap') {
+      const items = path.node.body.body;
+      for (const item of items) {
+        elements.add(item.typeAnnotation.typeAnnotation.typeName.name);
+      }
+    }
+  },
+});
+
+let augmentations = [];
+for (const element of elements) {
+  augmentations.push(`interface ${element} {
+  [symbol]: '${element}';
+}`);
+}
+
+const prefix = `//generated by scrips/build-augmentations.mjs\n`;
```

Comment:
Can you add to the comment block about why this is needed, how it's used, etc?

Thanks!